react/index.js

```diff
@@ -31,48 +34,55 @@ class Telemarketing extends Component {
     depersonify: PropTypes.func.isRequired,
     /** Mutation to impersonate a customer */
     impersonate: PropTypes.func.isRequired,
-    /** Mutation to initialize the session */
-    initializeSession: PropTypes.func.isRequired,
+    /** Query with the session */
+    session: PropTypes.shape({
+      loading: PropTypes.bool.isRequired,
+      getSession: PropTypes.shape({
+        adminUserEmail: PropTypes.string.isRequired,
+        adminUserId: PropTypes.string.isRequired,
+        impersonable: PropTypes.bool.isRequired,
+        profile: PropTypes.shape({
+          document: PropTypes.string,
+          phone: PropTypes.string,
+          firstName: PropTypes.string,
+          lastName: PropTypes.string,
+        }),
+        impersonate: PropTypes.shape({
+          storeUserId: PropTypes.string,
+          storeUserEmail: PropTypes.string,
+        }),
+      }),
+    }).isRequired,
   }
 
   state = {
     logged: false,
     loading: false,
     clientName: '',
     clientEmail: '',
+    emailInput: '',
     clientPhone: '',
     clientDocument: '',
     attendantEmail: '',
     canImpersonate: false,
+    isLoadingSession: true,
   }
 
-  componentDidMount = () => {
-    requestWithRetry(this.props.initializeSession)
-      .then(res => this.processSession(res.data && res.data.initializeSession))
-      .catch(err => console.log('err', err))
-  }
+  static getDerivedStateFromProps(props, state) {
+    const shouldProcessSession =
+      state.isLoadingSession === true && props.session.loading === false
```

Comment:
You can use `state.isLoadingSession && !props.session.loading`

Reply:
👍